### PR TITLE
feat(core, cli): builtin support of https and unix sockets

### PR DESCRIPTION
**Pull request by galvez**
TODO:

- [x] Review socket code
- [x] Adapt https PR into this
- [x] Write tests for sockets configurations
- [x] Write tests for https configurations

### CODE REVIEW

**Comment by galvez** [comment]
@pi0 @clarkdo followed your advice to the best of my judgment on https://github.com/nuxt/nuxt.js/pull/3812/files and https://github.com/nuxt/nuxt.js/pull/3403/files -- let me know if you have any additional feedback before I proceed to adding tests.

File: bin/nuxt-start [hunk 1]
```diff
@@ -77,8 +79,8 @@ if (nuxt.options.render.ssr === true) {
   }
 }
 
-const { port, host } = nuxt.options.server
+const { port, host, socket } = getLatestHost(argv)
```
Comment (pi0) [inline bin/nuxt-start]: Shouldn't we use it on `nuxt-dev` too?

File: lib/core/nuxt.js [hunk 2]
```diff
@@ -141,26 +142,52 @@ export default class Nuxt {
...
+      if (socket) {
+        args.path = socket
+      } else {
+        args.port = port
+        args.host = host
+      }
+
+      let appServer
+      const isHttps = !!this.options.http
+
```
[the remark this answers is not in the document]
Comment (galvez) [reply lib/core/nuxt.js]: @pi0 ops, that's a typo indeed!

File: lib/core/nuxt.js [hunk 3]
```diff
@@ -141,26 +142,52 @@ export default class Nuxt {
...
+      if (isHttps) {
+        const httpsOptions = this.options.https === true ? {} : this.options.http
```
[the remark this answers is not in the document]
Comment (galvez) [reply lib/core/nuxt.js]: @pi0 good call, thanks!

**Review by pi0 — APPROVED** [review]

**Pull request merged**